### ISSUE TITLE
The number of pages slot isn't correct when using the `renderToolbar`

### ISSUE THREAD

**Issue by phuoc-ng** [react-pdf-viewer/react-pdf-viewer#904]
```jsx
const renderToolbar = (
    Toolbar: (props: ToolbarProps) => React.ReactElement
) => (
    <Toolbar>
        {(toolbarSlot: ToolbarSlot) => {
            const { CurrentPageLabel, NumberOfPages } = toolbarSlot;
            return (
                <div data-testid="current-page-label">
                    <CurrentPageLabel /> of <NumberOfPages />
                </div>
            );
        }}
    </Toolbar>
);

const defaultLayoutPluginInstance = defaultLayoutPlugin({
    renderToolbar,
});
```

We will see that the `<NumberOfPages />` is empty when scrolling to the second page.

**Comment by ksaiteja19** [comment]
Work around for this issue is:

```
const [numberOfPages, setNumberOfPages] = useState(null);

<Toolbar>
{
    (tprops) => {
        const {
            CurrentPageInput,
            GoToNextPage,
            GoToPreviousPage
        } = tprops;
        
        return (
            <div className="pdf-header">
                <div className="view-page">
                    <GoToPreviousPage />
                    <div className="page-num">
                        <CurrentPageInput />
                        <div className="divider">/</div>
                        <span>{numberOfPages}</span>
                    </div>
                    <GoToNextPage />
                </div>
            </div>
        )
    }
}
</Toolbar>

<Viewer
    fileUrl={pdfURL}
    plugins={[toolbarPluginInstance]}
    onDocumentLoad={(d) => {
        setNumberOfPages(d.doc._pdfInfo.numPages)
    }}
/>

```

**Comment by phuoc-ng** [comment]
@ksaiteja19 It was fixed in v3.0.0.
